test/utils/describeConformance.tsx

```diff
@@ -898,11 +986,16 @@ export default function describeConformance(
         only.indexOf(testKey) !== -1 && skip.indexOf(testKey as keyof typeof fullSuite) === -1,
     ) as (keyof typeof fullSuite)[];
 
-    const slotBasedTests = ['slotsProp', 'slotPropsProp'];
-
     if (!slots) {
       // if `slots` are not defined, do not run tests that depend on them
-      filteredTests = filteredTests.filter((testKey) => !slotBasedTests.includes(testKey));
+      filteredTests = filteredTests.filter(
+        (testKey) => !['slotsProp', 'slotPropsProp'].includes(testKey),
+      );
+    }
+
+    if (!components) {
+      // if `components` are not defined, do not run tests that depend on them
+      filteredTests = filteredTests.filter((testKey) => !['componentsPropsProp'].includes(testKey));
```

Comment:
What about `componentsProp`?

Reply:
Those tests are in `function testMaterialUIComponentsPropsProp()`.

Reply:
I guess we will need another big refactoring but that can be done in a separate PR.